contracts/switchboard/default-switchboards/FastSwitchboard.sol

```diff
@@ -62,41 +72,63 @@ contract FastSwitchboard is SwitchboardBase {
     /**
      * @dev Function to attest a packet
      * @param packetId_ Packet ID
+     * @param proposalId_ Proposal ID
      * @param signature_ Signature of the packet
      */
-    function attest(bytes32 packetId_, bytes calldata signature_) external {
+    function attest(
+        bytes32 packetId_,
+        uint256 proposalId_,
+        bytes calldata signature_
+    ) external {
         uint32 srcChainSlug = uint32(uint256(packetId_) >> 224);
+        bytes32 root = socket__.packetIdRoots(packetId_, proposalId_);
+        if (root == bytes32(0)) revert InvalidRoot();
```

Comment:
Should we also take root as input and compare with the value from socket?

Reply:
I feel if packetId and proposalId uniquely identify a root, we don't need root as input.  It will mostly help in cases where there are issues in indexing on watcher's end such that he have correct root but wrong proposalId, and he ends up attesting for wrong proposalId . Although this can be avoided by watcher by checking root stored in packetIdRoots before attesting.

Reply:
There is the edge case introduced due to forks, maybe the watcher was on a fork while confirming but then another root gets proposed with a fork.
Very off case, that may also be saved by tripping.